### PR TITLE
evm: TransferSent(bytes32), fixes #503

### LINKED ISSUE #503: Emit NTT Manager digest in the TransferSent event
Emitting the NTT Manager digest in the `TransferSent` event can help off-chain systems such as indexers tie transfers across source/destination chains more easily.

Currently, 3rd party systems need to manually collect and properly serialize the NTT Manager message themselves to produce the digest, which requires a significant amount of code

### PR DESCRIPTION
**Pull request by evan-gray**
Before
```
| NttManager                            |   23,615 |        961 |
| NttManagerNoRateLimiting              |   18,169 |      6,407 |
```
After
```
| NttManager                            |   23,755 |        821 |
| NttManagerNoRateLimiting              |   18,403 |      6,173 |
```

### CODE REVIEW

**Review by djb15 — APPROVED** [review]
Nice test!

**Review by nik-suri — APPROVED** [review]

File: evm/test/IntegrationStandalone.t.sol [hunk 1]
```diff
@@ -184,8 +184,18 @@ contract TestEndToEndBase is Test, IRateLimiterEvents {
 
         vm.stopPrank();
 
+        // Get the TransferSent(bytes32) event to ensure it matches up with the TransferRedeemed(bytes32) event later
+        Vm.Log[] memory recordedLogs = vm.getRecordedLogs();
+        bytes32 sentEventDigest;
+        for (uint256 i = 0; i < recordedLogs.length; i++) {
+            if (recordedLogs[i].topics[0] == keccak256("TransferSent(bytes32)")) {
+                sentEventDigest = recordedLogs[i].topics[1];
```
Comment (banescusebi) [inline evm/test/IntegrationStandalone.t.sol:192]: Should there be a `break` out of the loop once this assignment happens?

File: evm/test/IntegrationStandalone.t.sol [hunk 2]
```diff
@@ -209,6 +219,19 @@ contract TestEndToEndBase is Test, IRateLimiterEvents {
             );
         }
 
+        // Get the TransferRedeemed(bytes32) event to ensure it matches up with the TransferSent(bytes32) event earlier
+        recordedLogs = vm.getRecordedLogs();
+        bytes32 recvEventDigest;
+        for (uint256 i = 0; i < recordedLogs.length; i++) {
+            if (recordedLogs[i].topics[0] == keccak256("TransferRedeemed(bytes32)")) {
+                recvEventDigest = recordedLogs[i].topics[1];
```
Comment (banescusebi) [inline evm/test/IntegrationStandalone.t.sol:227]: Should there be a `break` after this assignment?

**Review by banescusebi — COMMENTED** [review]
just a couple of nits

File: evm/test/IntegrationStandalone.t.sol (hunk 1 above)
Comment (evan-gray) [reply evm/test/IntegrationStandalone.t.sol:192]: fair. only in a test, so will cover in #522 if pushing again

File: evm/test/IntegrationStandalone.t.sol (hunk 2 above)
Comment (evan-gray) [reply evm/test/IntegrationStandalone.t.sol:227]: fair. only in a test, so will cover in #522 if pushing again

**Pull request merged**